frontend/store/companies/actions.js

```diff
@@ -0,0 +1,76 @@
+import types from "./mutation-types";
+
+export const fetchData = async (url) => {
+  const response = await fetch(url);
+  return response.json();
+};
+
+export default {
+  async fetchCompanies({ commit }) {
+    await fetchData("/api/faculties/wydzial-techniczny/companies.json").then(
+      (data) => {
+        commit(types.SET_COMPANIES, data);
+      }
+    );
+  },
```

Comment:
Use await or then, using them both is redundant. Stick to await, like this:
```js
  async fetchCompanies({ commit }) {
    const data = await fetchData("/api/faculties/wydzial-techniczny/companies.json");
    commit(types.SET_COMPANIES, data);
  },
  ```

Reply:
Okay, thanks.